tests/fixtures/scope-analysis/typeof-in-type-parameters.ts

```diff
@@ -0,0 +1,3 @@
+function g<T extends typeof g>(g: T): number {
```

Comment:
In `<T extends typeof g>`, is `g` the function name or the parameter name? Could one of the `g` identifiers be renamed for clarity, or are these supposed to be identical symbols in different symbol tables?

Reply:
The function name.

This test checks the parameter name doesn't shadow the function name for type parameters, as well.